framework/py/flwr/common/record/arrayrecord_test.py

```diff
@@ -328,6 +330,68 @@ def test_init_unrecognized_arg_raises_error(
         with self.assertRaisesRegex(TypeError, "Invalid arguments for ArrayRecord.*"):
             ArrayRecord(*args, **kwargs)
 
+    @parameterized.expand(  # type: ignore
+        [
+            ([np.array([1, 2]), np.array([3, 4])],),  # Two arrays
+            ([np.array(5)],),  # Single array
+            ([],),  # Empty list
+        ]
+    )
+    def test_inflation_deflation(self, array_content) -> None:
+        """Test inflation and deflation of ArrayRecord."""
+        arr_rec = ArrayRecord(array_content)
+
+        # Assert
+        # Expected children
+        assert len(arr_rec.children) == len(array_content)
```

Comment:
This is incorrect. There can be cases where `.children` has fewer elements than `array_content`.